WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -178,7 +187,13 @@ final class ProductSelectorViewModel: ObservableObject {
         guard let selectedProduct = products.first(where: { $0.productID == productID }) else {
             return
         }
-        if let onProductSelected = onProductSelected {
+
+        if supportsMultipleSelections, let onProductSelected {
+            // The selector supports multiple selection. Toggles the item, and triggers the selection
+            toggleSelection(productID: productID)
+            onProductSelected(selectedProduct)
+        } else if let onProductSelected {
+            // The selector supports single selection only
             onProductSelected(selectedProduct)
         } else {
             toggleSelection(productID: productID)
```

Comment:
I also missed the fact that coupons support multiple selections as well, but now the feature still work without setting `supportsMultipleSelections` to true for those selectors. This check is not reliable enough then, we should add a check for when `supportsMultipleSelections` is true too:

```suggestion
        } else if supportsMultipleSelections {
            toggleSelection(productID: productID)
```

Then when `supportsMultipleSelections` is false, no action is taken here.

That would require changing the initializer of the product selectors in `AddEditCouponViewModel` and `CouponRestrictionsViewModel` to set `supportsMultipleSelections` to `true` - would you mind updating that too? Thanks!

Reply:
Oh I see what you mean, I updated the view models here: https://github.com/woocommerce/woocommerce-ios/pull/8935/commits/58074731c6fb4314a58dbacc5037efd7d910c0bd and removed a unit test that will always fail, as we're not checking for row state on single selection anymore when toggling an item.